### ISSUE TITLE
Support Scaleway Kubernetes

### ISSUE THREAD

**Issue by Sh4d1** [opencost/opencost#560]
Hello! 

I'm wondering if you're open to contributions, I'd like to add Scaleway as a provider. If it's ok I'll start the implementation 😄

**Comment by dwbrown2** [comment]
Hi @Sh4d1, we're super supportive of creating new providers! Have any more context on what you would be looking to show for Scaleway?

@AjayTripathy can share more guidance too.

**Comment by Sh4d1** [comment]
Hey @dwbrown2 ! Thanks for the quick answer! 

I think a good starting point would be nodes/lb/block costs and then maybe out of cluster resources. 

A few gotchas though:
- we don't (at Scaleway) have a real billing API, just the price per node type
- there is no fix price /cpu and /ram, it depends on the offer
- we have different types of LBs, not just one

Then (I haven't read all the doc), I'm not sure if the focus should be on the managed offering or on a classical not managed cluster. I personally tend for at least start with the managed one. 

So basically I wanted to start and implement the provider interface as it seems it's only this part that needs new code right?

**Comment by AjayTripathy** [comment]
Hi @Sh4d1 . Very excited! This is one of the major reasons we are working to open source this project.

Yes, you only need to implement the provider interface-- and even then only a subset. A good place to start for understanding the minimal set to implement would be to look at the methods implemented by the CSVProvider, which simply takes a pricing sheet of nodes/persistentvolumes and matches them based on the ProviderID() field in kubernetes. 

https://github.com/kubecost/cost-model/blob/develop/pkg/cloud/csvprovider.go

In particular, you mainly need
DownloadPricingData() -- a threadsafe method for fetching your prices
NodePricing() -- the only requirement is to fill out the "Cost" field in returned Node. CPU/RAM/GPU breakdowns if they do not exist in the cloud provider will get set based on a ratio computed from a survey of other cloud providers.
PVPricing() -- the only requirement is to fill out the "Cost" field in the returned PV
(These costs fields are all hourly)

Loadbalancers are a pretty small proportion of people's cost, but if this is important to you we can discuss modifying the interface to pass in a k8s Service object and returning a cost-model Loadbalancer object. What's more likely is the service / k8s spec of a loadbalancer is quite thin and we'll need to join some additional information like a loadbalancer class/id somewhere to join with info on Scaleway.

My suggestion would be to start with DownloadPricingData() NodePricing() and PVPricing() for a v1 and e can discuss how important LB pricing is to your users.

**Comment by Sh4d1** [comment]
Thanks :+1: we are still misging a nice API to get the pricing though, I think I'll pause the implem until it's available :smile:

**Comment by aellert** [comment]
Hi @Sh4d1,
I hope Scaleway is still working to get Kubecost integration, I would love to use Kubecost with Scaleway :)

**Comment by dwbrown2** [comment]
Would love to see an integration here too! @Sh4d1 anything we can get you guys? One option would be to use a custom pricing sheet if there's still no billing APIs available for Scaleway.

**Comment by Sh4d1** [comment]
So just drafted this above ^ , feel free to test it :smile: as I said in the PR, it's still a hacky solution but it should work

**Comment by dwbrown2** [comment]
Thanks so much for the contribution here, @Sh4d1! We're just getting back from OSS summit and in final phases of a release. Will get you as review here as soon as we can.

**Comment by noreddinelam** [comment]
@Sh4d1 Hi, can we currently use the Scaleway provider to gather cloud costs, or is it not supported yet? 
If supported, what would the cloud-integration.json file need to look like, and are there additional steps to make it work?

**Comment by Sh4d1** [comment]
@noreddinelam no clue 😄 I stopped working with scaleway and did not touch this repo since 4 years 😅
